src/module/combat/combat.js

```diff
@@ -552,6 +552,153 @@ export class CombatSFRPG extends Combat {
+        this.flags.sfrpg.enemies = enemyInfo[0];
+        this.flags.sfrpg.enemyXP = enemyInfo[1];
+
+        const challengeInfo = this.calculateChallenge();
+        this.flags.sfrpg.CR = challengeInfo[0].CR;
+        this.flags.sfrpg.CRXPbounds = [challengeInfo[0].minXP, challengeInfo[0].totalXP];
+        this.flags.sfrpg.Difficulty = challengeInfo[1];
+        this.flags.sfrpg.IndXP = challengeInfo[2];
+
+        // Debug info
+        console.log(this);
+        console.log("Starfinder | The active combat encounter has ".concat(this.flags.sfrpg.PCs.length, " PC(s) [APL ", this.flags.sfrpg.APL, "], ", "and ", this.flags.sfrpg.enemies.length, " hostile NPC(s) [CR ", this.flags.sfrpg.CR, "]."));
+        console.log("Starfinder | The active combat encounter difficulty is ".concat(this.flags.sfrpg.Difficulty, " and is worth ", this.flags.sfrpg.IndXP, " individual XP per player."));
+
+    }
+
+    calculateAPL() {
+        let playerCombatants = [];
+        let playerLevels = [];
+
+        // Find all player-owned PCs and get their levels
+        for (const combatant of this.combatants) {
+            if (combatant.players.length > 0) {
+                playerCombatants.push(combatant);
+                playerLevels.push(combatant.actor.system.details.level.value);
+            }
+        }
+
+        // Calculate APL
+        if (!playerCombatants.length) {
+            return [playerCombatants, 0];
+        } else if (playerCombatants.length < 4) {
+            return [playerCombatants, Math.round(this.average(playerLevels)) - 1];
+        } else if (playerCombatants.length > 5) {
+            return [playerCombatants, Math.round(this.average(playerLevels)) + 1];
+        } else {
+            return [playerCombatants, Math.round(this.average(playerLevels))];
+        }
+    }
+
+    calculateEnemyXP() {
+        let enemyCombatants = [];
+        let enemyXP = [];
+
+        // Find all player-owned PCs and get their levels
+        for (const combatant of this.combatants) {
+            if (combatant.isNPC) {
+                if (combatant.token.disposition < 0) {
+                    enemyCombatants.push(combatant);
+                    enemyXP.push(combatant.actor.system.details.xp.value); // Enemy XP values
+                }
+            }
+        }
+        return [enemyCombatants, enemyXP.reduce((partialSum, a) => partialSum + a, 0)];
+    }
+
+    calculateChallenge() {
+        const XPTable = CONFIG.SFRPG.XPTable;
+        const APL = this.flags.sfrpg.APL;
+        const numPlayers = this.flags.sfrpg.PCs.length;
+        const numEnemies = this.flags.sfrpg.enemies.length;
+        const XPtotal = this.flags.sfrpg.enemyXP;
+        const XPNoCR = {CR: 0, minXP: 0, totalXP: 0, indOneThree: 0, indFourFive: 0, indSixPlus: 0};
+
+        // Check that Players and NPCs are both present
+        if (!numPlayers) {
+            return [XPNoCR, `${game.i18n.format("SFRPG.Combat.Difficulty.Levels.NoPCs")}`, 0];
```

Comment:
These strings don't need to be template literals, you can just call `game.i18n.format`

Reply:
Fixed